design-proposals/klip-53-pull-queries-on-streams.md

```diff
@@ -0,0 +1,87 @@
+# KLIP 53 - Pull Queries on Streams
+
+**Author**: John Roesler (@vvcephei) | 
+**Release Target**: 0.22 | 
+**Status**: _In Discussion_ | 
+**Discussion**: _link to the design discussion PR_
+
+**tl;dr:** People should be able to issue pull (ad-hoc, self-terminating) queries on streams,
+           just as they can with tables.
+           
+## Motivation and background
+
+ksqlDB allows the creation of both pull queries (which are issued ad-hoc and run to completion like
+a typical database query) and push queries (which are long-running subscriptions that continuously).
+ksqlDB also has a dual table/stream data model. You can issue push queries on tables and streams, 
+and you can issue pull queries on tables, but you cannot currently issue pull queries on tables.
```

Comment:
> but you cannot currently issue pull queries on tables.

-> on streams?

Reply:
Oops. I somehow mixed up the most important word in the sentence.